wellcomecollection/wellcomecollection.org
# Dash: Upgrade to Node20, fix TS errors and upgrade packages

Relates to #10926 

## What does this change?

- Moves to Node 20 (updated `.nvmrc` and Dockerfile)
- Upgrade all packages to latest versions, it's mostly patches but why not.
- Use to same `tsconfig` as workspace, just overrides `composite` as it's not a referenced project. Fixed errors that became flaggable.
- Deletes `babel.config.js` so we use SWC by default instead, which [is the case since Next 12](https://nextjs.org/docs/architecture/nextjs-compiler). I was having issue with server-side vs client-side with Styled-components and had to tweak the configuration. I thought might as well remove it as I don't think it's necessary anymore. Added the required config to `next.config.js` instead and it fixed my rendering issue.

## How to test

Run it locally `yarn dev` and navigate back and forth, see if it renders nicely.

## How can we measure success?

I've already run `yarn deploy` locally and it seems to work just fine (https://dash.wellcomecollection.org/). I believe that's the only way we deploy the dashboard, nothing automated anywhere, so that's success enough for me.

## Have we considered potential risks?

None I can think of since it deployed successfully.
